error: expected expression (related to __attribute__((fallthrough)))

<!-- Before reporting: search existing issues and check the FAQ. -->

- `nvim --version`: 0.3.7
- Vim (version: ) behaves differently? n/a
- Operating system/version: OS X 10.9.5
- Terminal name/version: n/a
- `$TERM`: n/a

### Steps to reproduce using `nvim -u NORC`

Run cmake and then make

### Actual behaviour

Build failure. First there are many warnings of this type:

```
./src/nvim/buffer.c:3736:7: warning: declaration does not declare anything [-Wmissing-declarations]
      FALLTHROUGH;
      ^~~~~~~~~~~
./src/nvim/macros.h:169:22: note: expanded from macro 'FALLTHROUGH'
# define FALLTHROUGH __attribute__((fallthrough))
                     ^~~~~~~~~~~~~
```

Then there is an error:

```
./src/nvim/regexp_nfa.c:3263:7: error: expected expression
      FALLTHROUGH;
      ^
./src/nvim/macros.h:169:22: note: expanded from macro 'FALLTHROUGH'
# define FALLTHROUGH __attribute__((fallthrough))
                     ^
```

It builds fine on OS X 10.10 and later. I assume this fallthrough attribute is something that isn't supported by the older compilers found on 10.9 and earlier.

### Expected behaviour

Successful build

--- a/src/nvim/macros.h
+++ b/src/nvim/macros.h
@@ -163,7 +163,8 @@
 # define NVIM_HAS_ATTRIBUTE __has_attribute
 #endif
 
-#if NVIM_HAS_ATTRIBUTE(fallthrough)
+#if NVIM_HAS_ATTRIBUTE(fallthrough) \
+    && (!defined(__apple_build_version__) || __apple_build_version__ >= 7000000)
 # define FALLTHROUGH __attribute__((fallthrough))
 #else
 # define FALLTHROUGH

